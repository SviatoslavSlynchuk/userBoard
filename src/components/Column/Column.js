import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import UserInfo from '../UserInfo/UserInfo';
import Helpers from '../../helpers';

const StyledSection = styled.section`
    min-width: 200px;
`;

const StyledHeading = styled.h3`
    text-align: center;
`;

const Column = (props) => {
    const { title, items } = props;
    return (
        <StyledSection>
            <StyledHeading>{Helpers.capitalize(title)}</StyledHeading>
            {items.length > 0
                ? items.map((item, index) => (
                    <UserInfo
                        key={index}
                        info={item}
                        onMoveToNextColumn={props.onMoveToNextColumn}
                        onMoveToPrevColumn={props.onMoveToPrevColumn}
                    />
                ))
                : null
            }
        </StyledSection>
    );
};

Column.propTypes = {
    items: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    onMoveToNextColumn: PropTypes.func.isRequired,
    onMoveToPrevColumn: PropTypes.func.isRequired,
};

export default Column;
