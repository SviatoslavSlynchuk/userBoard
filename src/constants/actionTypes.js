export const SET_PERSONNEL_DATA = 'SET_PERSONNEL_DATA';
export const GET_PERSONNEL_DATA = 'GET_PERSONNEL_DATA';
export const SET_NEW_STATUS = 'SET_NEW_STATUS';

export const SET_FILTER_OPERATION = 'SET_FILTER_OPERATION';
export const SET_FILTER_VALUE = 'SET_FILTER_VALUE';

export const SET_STATE_FROM_LOCAL_STORAGE = 'SET_STATE_FROM_LOCAL_STORAGE';
