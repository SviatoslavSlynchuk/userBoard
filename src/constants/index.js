export const DATA_URL = 'https://randomuser.me/api/?nat=gb&results=5';
export const APPLIED = 'applied';
export const INTERVIEWING = 'interviewing';
export const HIRED = 'hired';

export const FILTER_TYPE_NAME = 'name';
export const FILTER_TYPE_CITY = 'city';
