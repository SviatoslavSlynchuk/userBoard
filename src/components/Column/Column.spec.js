import React from 'react';
import Column from './Column';
import renderer from 'react-test-renderer';
import * as constants from '../../constants';

describe('Filter', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Column
                items={[
                    {
                        name: {
                            first: 'a',
                            last: 'b',
                        },
                        status: constants.APPLIED,
                        id: '0',
                        thumbnail: 'test',
                    },
                ]}
                title={constants.APPLIED}
                onMoveToNextColumn={() => {}}
                onMoveToPrevColumn={() => {}}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
