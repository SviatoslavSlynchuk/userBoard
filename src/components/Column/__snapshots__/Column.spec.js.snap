// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<section
  className="sc-ifAKCX fjyNqJ"
>
  <h3
    className="sc-EHOje hgnFSS"
  >
    Applied
  </h3>
  <div
    className="sc-bdVaJa hRtkfL"
  >
    <img
      className="sc-htpNat ioGRbM"
      src="test"
    />
    <span
      className="sc-bxivhb eNcWVP"
    >
      A B
    </span>
    <div
      className="sc-bwzfXH SkWqn"
    >
      <span />
      <button
        onClick={[Function]}
      >
        â‡’
      </button>
    </div>
  </div>
</section>
`;
